src/pysdccc/_cli.py

```diff
@@ -0,0 +1,133 @@
+import contextlib
+import io
+import pathlib
+import subprocess
+import sys
+import tempfile
+import zipfile
+
+import httpx
+
+from pysdccc import _download, _runner
+
+try:
+    import click
+except ImportError as import_error:
+    msg = "Cli not installed. Please install using 'pip install pysdccc[cli]."
+    raise ImportError(msg) from import_error
+
+
+class UrlType(click.ParamType):
+    name = 'url'
+
+    def convert(self, value: str, param: click.Parameter | None, ctx: click.Context | None) -> httpx.URL:
+        try:
+            return httpx.URL(value)
+        except Exception as e:  # noqa: BLE001
+            self.fail(f'{value!r} is not a valid url: {e}', param, ctx)
+
+
+URL = UrlType()
+
+
+class ProxyType(click.ParamType):
+    name = 'proxy'
+
+    def convert(self, value: str, param: click.Parameter | None, ctx: click.Context | None) -> httpx.Proxy:
+        try:
+            return httpx.Proxy(value)
+        except Exception as e:  # noqa: BLE001
+            self.fail(f'{value!r} is not a valid proxy: {e}', param, ctx)
```

Comment:
my linter complains that "Either all return statements in a function should return an expression, or none of them should."
pls. add a raise

Reply:
if I add a `return None` here, my IDE complains that this code is unreachable, because `self.fail(...)` raises an exception itself

Reply:
As discussed, it is the task of the user to keep this in mind. Check if the implementation of fail is as expected in the used version of click. The risk is low.